receiver/hostmetricsreceiver/internal/scraper/processscraper/get_process_handles_windows.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build windows
+
+package processscraper // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper"
+
+import (
+	"context"
+	"fmt"
+	"unsafe"
+
+	"github.com/shirou/gopsutil/v4/process"
+	"go.opentelemetry.io/collector/featuregate"
+	"golang.org/x/sys/windows"
+)
+
+var useNewGetProcessHandles = featuregate.GlobalRegistry().MustRegister(
+	"hostmetrics.process.OnWindowsUseNewGetProcesses",
+	featuregate.StageBeta,
+	featuregate.WithRegisterDescription("If disabled, the scraper will use the legacy implementation to retrieve process handles."),
+)
+
+func getGopsutilProcessHandles(ctx context.Context) (processHandles, error) {
```

Comment:
I think this function needs to be somewhere else; this file is build constrained on `GOOS` being `windows`, but is then used in `process_scraper.go` which is not under the same constraint. Maybe move this to `process.go`?

Reply:
There is [`get_process_handles_others.go`](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38589/files#diff-e5aa3065e388b30939320aa1a375542f2a777b6029d8e6689038995c10941c63) that is the same as the old implementation and built for all non-Windows platforms.